### DISCUSSION TITLE
Beercss should have an prefix ? Or perhaps even an option to include or omit a prefix ?

### DISCUSSION THREAD

**Discussion by KicKerBNU** [beercss/beercss#131] [General]
Hello Beercss community 🍺 

Changing some emails with Beercss users and reading around our github page, many times i see people struggling with conflicts with differents frameworks and class conflicts between Beercss and other frameworks or even global classes...

We all know the best way to work safe and fine in a perfect world its working only with 1 framework in the whole project. But unfortunately we know the real world could be very complex and people have problems to erase old framework to work exclusively with a new one.

To help part of the community should we work on a workaround from this problem ?

**Comment by tmsnarr** [comment]
I would like to start implementing beer css in an existing project that doesn't currently use a UI library. In this project I use plain CSS but sometimes my classes clash with beer css classes causing issues. If there was an option to install a prefixed version of beer css classes (e.g. beer-container or whatever) then I could begin to implement it into my project iteratively without issues.

**Reply by KicKerBNU** [reply]
@leonardorafael what are your thoughts about it ? this gonna be a breaking change

**Reply by tmsnarr** [reply]
I wonder if it could be an additional option rather than a breaking change.

**Reply by laurentpayot** [reply]
I personally don’t have conflict issues, but if this problem has to be fixed with a prefix I think it should be in a separate bundle because this prefix-added bundle would be bigger than the standard one. A separate bundle would also avoid breaking changes for people who don’t need a prefix (most people I think).

**Comment by leonardorafael** [comment]
I have some legacy systems that I would like to use Beer CSS, too. But I can't see a nice way to do that, without hurts the rules and DX of Beer CSS project. Can we have another options instead prefix? Using prefix, will double size all CSS and HTML code. Maybe a root web component? I really don't know.

```html
// before prefix
<article>
  <h5>Title</h5>
  <p>Some text here</p>
  <nav>
    <button>Button</button>
  </nav>
</article>
...

// after prefix
<article class="beer-article">
  <h5 class="beer-h5">Title</h5>
  <p class="beer-p">Some text here</p>
  <nav class="beer-nav">
    <button class="beer-button">Button</button>
  </nav>
</article>
...

// after prefix (all div)
<div class="beer-article">
  <div class="beer-h5">Title</div>
  <div class="beer-p">Some text here</div>
  <div class="beer-nav">
    <div class="beer-button">Button</div>
  </div>
</div>
...

// root web component
// shadow dom will avoid CSS colision in component?
// the component can load the css/js automatically?
// can be a fullscreen mode or not?
<beer>
  <article>
    <h5>Title</h5>
    <p>Some text here</p>
    <nav>
      <button>Button</button>
    </nav>
  </article>
  ...
</beer>
```

I like the root web component approach. I don't know if this will work as expected, but seems a nice way to preserve the rules and DX of project. And this can preserve all ways to install the project too. We need to think about it.
